contributions/catalog/controllers/contribute.py

```diff
@@ -212,21 +219,140 @@ def contribution_edit(contribution_id):
         the_json_res = get_contribution(contribution_id)
         # check if the user is editable then set the is_editable
         is_editable = False
+        is_reviewer = False
         username = session["username"]
         headers = requestutil.get_header_using_session(session)
-        is_editable = adminutil.check_if_reviewer(username, headers)
+
+        # check if the user is in contribution's admins
+        if username in the_json_res["contributionAdmins"]:
+            is_editable = True
+
+        # check if the user is a reviewer
+        is_reviewer = adminutil.check_if_reviewer(username, headers)
+
+        if is_reviewer:
+            is_editable = True
 
         # get capability list to create required capability list
         required_capability_list = requestutil.request_required_capability_list(headers)
 
         if is_editable:
-            return render_template('contribute/contribute.html', required_capabilities=required_capability_list,
+            return render_template('contribute/contribute.html', is_reviewer=is_reviewer, required_capabilities=required_capability_list,
                                    is_editable=is_editable, user=session["name"],
                                    token=session['oauth_token']['access_token'], post=the_json_res)
         else:
             s = "You don't have a permission to edit the contribution."
             return render_template('contribute/error.html', error_msg=s)
 
+@bp.route('/contributions/<contribution_id>/review', methods=['GET', 'POST'])
+@login_required
+def contribution_review(contribution_id):
+    review_flag = request.args.get('review')
```

Comment:
What is the purpose of the `review` query parameter here?  Could we not just redirect users to `/contributions/<contribution_id>/review` page and display the error message of permission if they don't have permission to view it?

Reply:
it was like an insurance to check if this is really the review page. However, it is probably not needed because there shouldn't be any occasion that this flag makes any problem. I will remove those.

Reply:
the flag has been removed

Reply:
Okay. Thank you.